### PR TITLE
Paper: Echostack: A flexible and scalable open-source software toolbox for echosounder data processing

### PR DESCRIPTION
**Pull request by leewujung**
If you are creating this PR in order to submit a draft of your paper, please name your PR with `Paper: <title>`. An editor will then add a `paper` label and GitHub Actions will be run to check and build your paper.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#preview-your-paper) for more information.

_Editor_: Meghann Agarwal @mepa

_Reviewers_: 
- Andy Terrel @aterrel
- Ashwin Hegde @AshwinHegde

### CODE REVIEW

**Comment by leewujung** [comment]
One of the papers cited does not have a DOI, so the checks keep on failing. I also replaced a couple bibtex items for websites with direct URL (since they also don't have DOIs), but feel it may still be better to cite them as bibtex items, otherwise the same URL gets repeated more than once, which feels a little bizarre. 

@hongsupshin: Are these check failings ok, or could you suggest how to address them? Thanks!

**Update:** Just saw the option to explicitly ignore the `doi-exists` check in `myst.yml`, so this is resolved! :)

**Comment by ameyxd** [comment]
great, thanks @leewujung

**Comment by mepa** [comment]
Hi @chuchugo and @AshwinHegde! In case a little extra time is needed, the initial complete review deadline has been extended to next [Wednesday, July 3rd](https://github.com/scipy-conference/scipy_proceedings/blob/2024/README.md#instructions-for-reviewers).

**Comment by aterrel** [comment]
Howdy! I agreed to review this paper and will get a review in by July 8th.

**Comment by mepa** [comment]
> Howdy! I agreed to review this paper and will get a review in by July 8th.

Thank you, @aterrel!

**Comment by mepa** [comment]
> the initial complete review deadline has been extended to next [Wednesday, July 3rd](https://github.com/scipy-conference/scipy_proceedings/blob/2024/README.md#instructions-for-reviewers).

Update: the Proceedings Committee has extended the initial complete review deadline to Monday, July 8th.

**Comment by aterrel** [comment]
Independent Review Report
=========================


Reviewer: Andy R. Terrel

Department/Center/Division: Compute Products

Institution/University/Company: NVIDIA

Field of interest / expertise: Computer Science / Computational Math

Country: USA


Article reviewed: Echostack: A flexible and scalable open-source software toolbox for echosounder data processing

GENERAL EVALUATION
------------------

Please rate the paper using the following criteria (please use the abbreviation
to the right of the description)::

  below     doesn't meet standards for academic publication
  meets     meets or exceeds the standards for academic publication
  n/a       not applicable

- Quality of the approach: meets

- Quality of the writing: meets

- Quality of the figures/tables: meets


SPECIFIC EVALUATION
-------------------

- Is the code made publicly available and does the article sufficiently
  describe how to access it?  We aim not to publish papers that essentially
  advertise proprietary software.  Therefore, if the code is not publicly
  available, please provide a one- to two- sentence response to each of the
  following questions:

Yes

- Does the article present the problem in an appropriate context?
  Specifically, does it:

 Yes this was a well written exposition of the stack needed for modern scientific computing. As someone who has built many systems of this nature, I think the authors did a marvelous job motivating and explaining all the pieces.

- Is the content of the paper accessible to a computational scientist
  with no specific knowledge in the given field?

Yes

- Does the paper describe a well-formulated scientific or technical
  achievement?

Yes, creating a usable software stack for this domain is an accomplishment.

- Are the technical and scientific decisions well-motivated and
  clearly explained?

Yes

- Are the code examples (if any) sound, clear, and well-written?

No code samples

- Is the paper factually correct?

Yes

- Is the language and grammar of sufficient quality?

Yes

- Are the conclusions justified?

Yes

- Is prior work properly and fully cited?

I'm not an expert on the domain but the citations were plentiful for me to follow the factual reasoning

- Should any part of the article be shortened or expanded? Please explain.

No, it is just right.

- In your view, is the paper fit for publication in the conference proceedings?
  Please suggest specific improvements and indicate whether you think the
  article needs a significant rewrite (rather than a minor revision).

Yes it is fit for publication.

File: papers/wu-jung_lee/main.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
...ools for flexible and scalable access, organization, processing, and visualization of echosounder data. Echostack is a domain-specific adoption of the core libraries in the Pandata stack, such as Zarr, Xarray, and Dask [@bednar2023]. It streamlines the composition and execution of common echosounder data workflows, thereby allowing researchers to focus on the key interpretive stage of scientific data analysis. As echosounder data processing frequently requires combining heterogeneous data sources and models in addition to regular array operations, we designed the Echostack packages with modularity in mind, to 1) enable and facilitate broader code reuse, especially for routine processing steps that are often shared across echosounder data workflows [@haris2021], and 2) promote flexible integration of domain-specific operations with more general analytical tools, such as machine learning (ML) libraries. The Echostack also provides a friendly on-ramp for researchers who are not already familiar with the scientific Python software ecosystem but possess domain expertise and can quickly benefit from "out-of-the-box" capabilities such native cloud access and distributed computing support.
```
Comment (AshwinHegde) [inline papers/wu-jung_lee/main.md]: Minor typo - replace "... capabilities such native cloud access ..." with "... capabilities such _as_ native cloud access ..."

File: papers/wu-jung_lee/main.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+:::{figure} fig_usecase_ship2cloud.png
+:label: fig:usecase_ship2cloud
+:width: 700px
+:align: center
+Use case 2: A ship-to-cloud ML pipeline.
+:::
+
+
+### Use case 3: Mass production of cloud-optimized, analysis-ready (ARCO) data products
```
Comment (AshwinHegde) [inline papers/wu-jung_lee/main.md]: Minor typo - should the title read "... _analysis-ready, cloud-optimized_ (ARCO) ..." instead?

**Review by AshwinHegde — COMMENTED** [review]
# Review

Paper reviewed: Echostack: A flexible and scalable open-source software toolbox for echosounder data processing
Reviewer: Ashwin Hegde

The paper introduces a suite of open source python packages, collectively called Echostack, that can be used for echosounder data processing, analysis, visualization, and workflow orchestration. The development is an ongoing effort and the paper serves as an interim report that outlines the motivation, describes the current capabilities of the set of tools and proposes future development directions.

The paper is very well written. The motivation for the work is clearly outlined upfront with clear pain points in existing workflows identified. The presentation of design principles for the software packages is appreciated. The solution is also introduced well. Even though I don't work in this domain, the presentation was easy to follow. The authors outline the functionalities of each of the packages well, also providing example use cases integrating different subsets of them. It would have been useful to provide code examples/tutorials as part of each of the package repositories but this is promised as part of future work. All figures are clear, captioned, and referenced correctly.

The work presented here seems highly relevant and directly impactful to the fisheries acoustics, ocean sciences, and environmental science communities. Since the packages are built in a modular way, some of these could be extended to broader disciplines, eg: geology and geophysics, archaelogy, hydrology, etc. Usage of the software is explained well and augmented with multiple examples. All the packages have separate repositories with easy to follow installation instructions for different environments.

The authors provide permissive open licenses for the packages and encourage open contribution which is great for the community. 

Overall, I believe the Echostack suite is a significant advance in the processing, analysis, and visualization of echosounder data. The authors have done a good job of presenting the motivations, and proposing a well-structured solution. Apart from what I believe to be minor typos, I recommend publication without modification.

**Review by aterrel — APPROVED** [review]

File: papers/wu-jung_lee/main.md (hunk 1 above)
Comment (leewujung) [reply papers/wu-jung_lee/main.md]: Thanks! Added!

File: papers/wu-jung_lee/main.md (hunk 2 above)
Comment (leewujung) [reply papers/wu-jung_lee/main.md]: Yes, thanks for catching this. Added!

**Comment by leewujung** [comment]
Thanks very much @aterrel and @AshwinHegde for reviewing this article! I have corrected the typos you have noted, and will do another couple thorough reads to make sure any other small things get caught.

**Comment by mepa** [comment]
Hi @aterrel and @AshwinHegde - Do you feel that this paper in its current form is ready for inclusion in the Proceedings?

The final reviewer decision deadline is September 9th so I'm just checking in to confirm reviewer decisions as that deadline approaches.

**Comment by mepa** [comment]
Reviewer sign-off is inferred from statements in the comments above.
From @aterrel:
> "Yes it is fit for publication."

From @AshwinHegde:
> "Apart from what I believe to be minor typos, I recommend publication without modification."

Thank you both for reviewing!

**Pull request merged**